### PR TITLE
[Fundamental] Add GH workflow to update version and add tag for promptflow

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Add a workflow to update version and add tag for release branch.

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/workflows/create_promptflow_release_tag.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Create promptflow release tag
+
+on:
+  workflow_dispatch:
```
Comment (D-W-) [inline .github/workflows/create_promptflow_release_tag.yml:4]: Shall we restrict this to only run on release branch(since it will modify _version.py)?

Comment (zhengfeiwang) [reply .github/workflows/create_promptflow_release_tag.yml:4]: good catch, add a check on prefix

File: .github/workflows/create_promptflow_release_tag.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: create branch
+        run: |
+          set -x -e
+
+          branch_name=$(git rev-parse --abbrev-ref HEAD)
+          echo "branch name: $branch_name"
+          release_version=${branch_name#release/promptflow/}
+          echo "release version: $release_version"
```
Comment (crazygao) [inline .github/workflows/create_promptflow_release_tag.yml:27]: Should Fail if release version doesn't match the pattern.

Also git rev-parse is getting SHA?

Comment (zhengfeiwang) [reply .github/workflows/create_promptflow_release_tag.yml:27]: latest commit have resolved these

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**